src/main/resources/initDB/schema.sql

```diff
@@ -1,5 +1,125 @@
+        );
+    END IF;
+END' language plpgsql;
+
+CREATE TABLE IF NOT EXISTS orders(order_id BIGSERIAL PRIMARY KEY,
+                    client_id BIGINT NOT NULL,
+                    address VARCHAR(100) NOT NULL,
+                    coordinates geometry NOT NULL,
+                    warehouse_id BIGINT,
+                    courier_id BIGINT, 
+                    status order_status NOT NULL DEFAULT 'created',
+                    date_start TIMESTAMP NOT NULL,
+                    date_end TIMESTAMP,
+                    overall_cost NUMERIC(7, 2) NOT NULL CHECK(overall_cost > 0), 
+                    high_demand_coeff NUMERIC(3, 2) NOT NULL DEFAULT 1 CHECK(high_demand_coeff >= 1 AND high_demand_coeff < 3),
+                    discount NUMERIC(7, 2) NOT NULL DEFAULT 0 CHECK(discount >= 0),
+                    promo_code_id BIGINT,
+                    client_rating NUMERIC(3, 2) CHECK((client_rating IS NULL) OR (client_rating BETWEEN 0 AND 5)),
+                    delivery_rating NUMERIC(3, 2) CHECK((delivery_rating IS NULL) OR (delivery_rating BETWEEN 0 AND 5)),
+                    FOREIGN KEY(warehouse_id) REFERENCES warehouses(warehouse_id) ON DELETE CASCADE,
+                    FOREIGN KEY(client_id) REFERENCES clients(client_id) ON DELETE CASCADE,
+                    FOREIGN KEY(courier_id) REFERENCES couriers(courier_id) ON DELETE SET NULL,
+                    FOREIGN KEY(promo_code_id) REFERENCES promo_codes(promo_code_id) ON DELETE CASCADE);
+
+DROP FUNCTION IF EXISTS warehouse_update_is_deactivated() CASCADE;
+CREATE OR REPLACE FUNCTION warehouse_update_is_deactivated() RETURNS TRIGGER
+AS '
+BEGIN
+        IF NEW.is_deactivated = true THEN
+            UPDATE couriers SET warehouse_id = NULL WHERE warehouse_id = NEW.warehouse_id;
+            UPDATE moderators SET warehouse_id = NULL WHERE warehouse_id = NEW.warehouse_id;
+            UPDATE orders SET warehouse_id = NULL WHERE warehouse_id = NEW.warehouse_id;
+        END IF;
+    RETURN NEW;
+END;
+' LANGUAGE plpgsql;
+
+CREATE TRIGGER warehouse_update_is_deactivated
+    AFTER UPDATE OF is_deactivated
+    ON warehouses
+    FOR EACH ROW
+    EXECUTE FUNCTION warehouse_update_is_deactivated();
+
+    
+DROP FUNCTION IF EXISTS order_update_status() CASCADE;
+CREATE OR REPLACE FUNCTION order_update_status() RETURNS TRIGGER
+AS '
+BEGIN
+        IF NEW.status = ''cancelled'' THEN
+            DECLARE pos BIGINT;
+            BEGIN
+                FOR pos IN (SELECT product_position_id FROM (SELECT * FROM orders INNER JOIN orders_product_positions USING(order_id) WHERE order_id = NEW.order_id) positions) LOOP
+                UPDATE product_positions orders_product_positions SET current_amount = current_amount + (SELECT amount FROM orders INNER JOIN orders_product_positions USING(order_id) WHERE order_id = NEW.order_id AND product_position_id = pos) WHERE product_position_id = pos; 
```

Comment:
Тут случайно нет ошибки? может нужно убрать **orders_product_positions** в начале ?  
```suggestion
                UPDATE product_positions SET current_amount = current_amount + (SELECT amount FROM orders INNER JOIN orders_product_positions USING(order_id) WHERE order_id = NEW.order_id AND product_position_id = pos) WHERE product_position_id = pos; 
```